seastar_cmake.py

```diff
@@ -21,10 +21,15 @@
 
 ROOT_PATH = os.path.realpath(os.path.dirname(__file__))
 
-BUILD_PATHS = { mode: os.path.join(ROOT_PATH, 'build', mode) for mode in SUPPORTED_MODES }
-
 COOKING_BASIC_ARGS = ['./cooking.sh']
 
+def build_path(mode, build_root = 'build'):
+    """Return the absolute path to the build directory for the given mode,
+    i.e., seastar_dir/<build_root>/<mode>"""
+    if mode not in SUPPORTED_MODES:
+        raise RuntimeError('Mode ' + mode + ' is not a supported mode')
+    return os.path.join(ROOT_PATH, build_root, mode)
```

Comment:
This is the core of the change: instead of hardcoding `build` as the root build directory, we use the `build_root` parameter, which still defaults to `build`. Nobody should see any change unless they specify `--build-root` on the command line.

I use it to do something like:

```
./configure.py --build-root=build-clang-12 --compiler=clang++-12 --c-compiler=clang-12 --cflags='-fdebug-default-version=4 -march=skylake'
```

So I can have separate build dirs for gcc, clang-12 and clang-14.